src/extension/background-script/actions/ln/sendPayment.js

```diff
@@ -42,28 +42,31 @@ async function sendPaymentWithAllowance(
   return response;
 }
 
-async function sendPaymentWithPrompt(message, paymentRequestDetails) {
+async function payWithPrompt(message) {
+  await utils.openPrompt(message);
+}
+
+export async function weblnPay(message) {
+  const { paymentRequest } = message.args;
   const connector = state.getState().getConnector();
-  let promptResponse;
-  let paymentResponse;
+  const paymentRequestDetails = parsePaymentRequest({
+    request: paymentRequest,
+  });
+
   try {
-    promptResponse = await utils.openPrompt(message);
-    if (promptResponse && promptResponse.data.confirmed) {
-      paymentResponse = await connector.sendPayment({
-        paymentRequest: message.args.paymentRequest,
-      });
-      publishPaymentNotification(
-        message,
-        paymentRequestDetails,
-        paymentResponse
-      );
-      return paymentResponse;
-    }
+    const response = await connector.sendPayment({
+      paymentRequest,
+    });
+    publishPaymentNotification(
+      message.args.message,
+      paymentRequestDetails,
+      response
+    );
+
+    return response;
```

Comment:
I think this new flow is better, isn't it? 
better than expecting a respone from the openPrompt and then continuing based on that response

what do you think?